waku/node/waku_node.nim

```diff
@@ -444,9 +443,15 @@ proc mountFilter*(
   )
 
   if node.started:
-    await node.wakuFilter.start()
+    try:
+      await node.wakuFilter.start()
+    except CatchableError:
+      error "failed to start wakuFilter", error = getCurrentExceptionMsg()
```

Comment:
You don't like the `catch: ` macro?
```suggestion
    let catchable = catch: await node.wakuFilter.start()
    if catchable.isErr():
      error "failed to start wakuFilter", error = catchable.error.msg
```

Reply:
@SionoiS We would need better a valueOrCatch macro... hm, due I don't think catch works together with valueOr template.

Reply:
> @SionoiS We would need better a valueOrCatch macro... hm, due I don't think catch works together with valueOr template.

Yes it's unfortunate, otherwise this would be possible
```
(catch: await node.wakuFilter.start()).isOkOr:
   error "failed to start wakuFilter", error = error.msg
```

Reply:
> > @SionoiS We would need better a valueOrCatch macro... hm, due I don't think catch works together with valueOr template.
> 
> Yes it's unfortunate, otherwise this would be possible
> 
> ```
> (catch: await node.wakuFilter.start()).isOkOr:
>    error "failed to start wakuFilter", error = error.msg
> ```

Well we can add! I may do it somewhere.

Reply:
> You don't like the `catch: ` macro?

Well, I "don't like" maybe because I don't fully understand it xD
To me, it sounds a bit more verbose the traditional approach but I'll give it another glance ;P